plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponse.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.ml.processor;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.search.SearchResponseSections;
+import org.opensearch.action.search.ShardSearchFailure;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+public class MLInferenceSearchResponse extends SearchResponse {
+    private static final String EXT_SECTION_NAME = "ext";
+
+    private Map<String, Object> params;
+
+    public MLInferenceSearchResponse(
+        Map<String, Object> params,
+        SearchResponseSections internalResponse,
+        String scrollId,
+        int totalShards,
+        int successfulShards,
+        int skippedShards,
+        long tookInMillis,
+        ShardSearchFailure[] shardFailures,
+        Clusters clusters
+    ) {
+        super(internalResponse, scrollId, totalShards, successfulShards, skippedShards, tookInMillis, shardFailures, clusters);
+        this.params = params;
+    }
+
+    public void setParams(Map<String, Object> params) {
+        this.params = params;
+    }
+
+    public Map<String, Object> getParams() {
+        return this.params;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        innerToXContent(builder, params);
```

Comment:
Hey Dhrubo, From my experience when you are trying to get the `_source` Map<String, Object> object into the format the search response has you need to to convert it to xContent for it to go back. Here is a snippet of program I did

```
                XContentBuilder builder = XContentBuilder.builder(mediaTypeAndSourceMapTuple.v1().xContent());
                builder.map(sourceAsMap);
                hit.sourceRef(BytesReference.bytes(builder));
```
My hunch is that since this new type of search response is doing a different interpretation of the object it will invoke this method when it converts it back.
I say this because of this line ` MLInferenceSearchResponse extends SearchResponse` and if you check the processor code it creates a `MLInferenceSearchResponse` not the usual `SearchResponse`